mlrun/api/crud/model_monitoring/_model_endpoint_stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,551 @@
+                db.String(255),
+            ),
+            db.Column(model_monitoring_constants.EventFieldType.ACTIVE, db.Boolean),
+            db.Column(
+                model_monitoring_constants.EventFieldType.MONITORING_MODE,
+                db.String(10),
+            ),
+            db.Column(model_monitoring_constants.EventFieldType.FEATURE_STATS, db.Text),
+            db.Column(model_monitoring_constants.EventFieldType.CURRENT_STATS, db.Text),
+            db.Column(model_monitoring_constants.EventFieldType.FEATURE_NAMES, db.Text),
+            db.Column(model_monitoring_constants.EventFieldType.CHILDREN, db.Text),
+            db.Column(model_monitoring_constants.EventFieldType.LABEL_NAMES, db.Text),
+            db.Column(model_monitoring_constants.EventFieldType.TIMESTAMP, db.DateTime),
+            db.Column(
+                model_monitoring_constants.EventFieldType.ENDPOINT_TYPE,
+                db.String(10),
+            ),
+            db.Column(model_monitoring_constants.EventFieldType.CHILDREN_UIDS, db.Text),
+            db.Column(
+                model_monitoring_constants.EventFieldType.DRIFT_MEASURES, db.Text
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.DRIFT_STATUS,
+                db.String(40),
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.MONITOR_CONFIGURATION,
+                db.Text,
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.FEATURE_SET_URI,
+                db.String(255),
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.FIRST_REQUEST,
+                db.String(40),
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.LAST_REQUEST,
+                db.String(40),
+            ),
+            db.Column(
+                model_monitoring_constants.EventFieldType.ERROR_COUNT, db.Integer
+            ),
+            db.Column(model_monitoring_constants.EventFieldType.METRICS, db.Text),
+        )
+
+    def delete_model_endpoints_resources(
+        self, endpoints: mlrun.api.schemas.model_endpoints.ModelEndpointList
+    ):
+        """
+        Delete all model endpoints resources in both SQL and the time series DB. In addition, delete the model
+        endpoints table from SQL if it's empty at the end of the process.
+
+        :param endpoints: An object of `ModelEndpointList` which is literally a list of model endpoints along with some
+                          metadata. To get a standard list of model endpoints use `ModelEndpointList.endpoints`.
+        """
+
+        # Delete model endpoint record from SQL table
+        for endpoint in endpoints.endpoints:
+            self.delete_model_endpoint(
+                endpoint.metadata.uid,
+            )
+
+        # Drop the SQL table if it's empty
+        self._drop_table()
```

Comment:
I wouldn't drop the table every time it's empty, because this may result in many create/delete calls, and if the table is empty then it's not really costly to keep it. Dropping the table should happen only if we completely disable monitoring or something (which is a flow that I don't think exists).

Reply:
Alright, even though I think that as part of the delete project api it does make sense to delete this table if it's empty. In addition, if the table is empty and we decide to change the model endpoint schema then we won't be able to enter a new model endpoint record since the table is based on the old schema. In that case, the user will have to modify/delete the table manually.